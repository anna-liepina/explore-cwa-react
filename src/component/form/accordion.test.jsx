import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Accordion from './accordion';

describe('<Accordion/>', () => {
    const props = {
    };

    const optionalProps = {
        /** should be passed 'as is' */
        'data-cy': 'optProps.data-cy',
        className: 'optProps.className',
        label: 'optProps.label',
        errors: ['optProps.errors[0]', 'optProps.errors[1]'],
        isCollapsed: true,
        children: <div className="optionalProps.children" />
    };

    const expandedlProps = {
        ...optionalProps,
        isCollapsed: false,
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment } = render(<Accordion {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props - collapsed', () => {
            const { asFragment } = render(<Accordion {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });
        it('with optional/required props - expanded', () => {
            const { asFragment } = render(<Accordion {...props} {...expandedlProps} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
