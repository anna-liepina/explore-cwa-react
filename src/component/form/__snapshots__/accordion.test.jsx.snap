// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion/> render with default/required props 1`] = `
<DocumentFragment>
  <section
    class="accordion "
  />
</DocumentFragment>
`;

exports[`<Accordion/> render with optional/required props - collapsed 1`] = `
<DocumentFragment>
  <section
    class="accordion--collapsed optProps.className"
  />
</DocumentFragment>
`;

exports[`<Accordion/> render with optional/required props - expanded 1`] = `
<DocumentFragment>
  <section
    class="accordion optProps.className"
  >
    <div
      class="optionalProps.children"
    />
  </section>
</DocumentFragment>
`;
